typescript/src/schema/proto.ts

```diff
@@ -33,6 +33,19 @@ const COMMON_PROTO_LIST = [
   'google.type',
 ];
 
+const STRICT_MODE_RESERVED_WORDS = new Set<String>([
```

Comment:
Sorry, this cannot be here since it is specific for TypeScript and the generator code must be language independent by design (otherwise, all possible issues will arise).  The only possible place for this logic is the `namer.js` in the template. Good news is that it should be pretty straightforward to move it there.

Reply:
Thanks for the advice. That's smart design. New lesson learned.